java/ql/src/utils/model-generator/internal/CaptureModelsSpecific.qll

```diff
@@ -223,24 +225,21 @@ predicate isOwnInstanceAccessNode(ReturnNode node) {
 }
 
 /**
- * Language specific parts of the `PropagateToSinkConfiguration`.
+ * Holds if `source` is an api entrypoint relevant for creating sink models.
  */
-class PropagateToSinkConfigurationSpecific extends TaintTracking::Configuration {
-  PropagateToSinkConfigurationSpecific() { this = "parameters or fields flowing into sinks" }
-
-  override predicate isSource(DataFlow::Node source) {
-    (
-      source.asExpr().(J::FieldAccess).isOwnFieldAccess() or
-      source instanceof DataFlow::ParameterNode
-    ) and
-    source.getEnclosingCallable().isPublic() and
-    exists(J::RefType t |
-      t = source.getEnclosingCallable().getDeclaringType().getAnAncestor() and
-      not t instanceof J::TypeObject and
-      t.isPublic()
-    ) and
-    isRelevantForModels(source.getEnclosingCallable())
-  }
+predicate apiSource(DataFlow::Node source) {
+  (
+    source.asExpr().(J::FieldAccess).isOwnFieldAccess() or
+    source instanceof DataFlow::ParameterNode
+  ) and
+  source.getEnclosingCallable().isPublic() and
+  exists(J::RefType t |
+    t = source.getEnclosingCallable().getDeclaringType().getAnAncestor() and
+    not t instanceof J::TypeObject and
+    t.isPublic()
+  ) and
+  isRelevantForModels(source.getEnclosingCallable()) and
+  exists(asPartialModel(source.getEnclosingCallable()))
```

Comment:
What is the impact of already narrowing the potential sources nodes by our ability to print a partial model for its enclosing callable (I know this is needed for the model - but which potential source nodes does this exclude?)?

Reply:
Not sure, I didn't check that thoroughly - I only noticed that it made a (somewhat significant) difference.